### PR TITLE
add support for %{shrink:}, derived from rpm5.org

### PR DESCRIPTION
**Pull request by proyvind**
Shrinks body by removing all leading and trailing whitespaces and
reducing intermediate whitespaces to a single space character.

### CODE REVIEW

File: rpmio/macro.c [hunk 1]
```diff
@@ -864,6 +865,28 @@ doFoo(MacroBuf mb, int negate, const char * f, size_t fn,
 	if ((b = strrchr(buf, '/')) != NULL)
 	    *b = '\0';
 	b = buf;
+    } else if (STREQ("shrink", f, fn)) {
+	/*
+	 * shrink body by removing all leading and trailing whitespaces and
```
Comment (ignatenkobrain) [inline rpmio/macro.c]: isn't this `trim` or `strip`?

**Comment by proyvind** [comment]
There's no %{trim:} nor %{strip:}.. ;)
But beyond just trimming/stripping, it shrinks it all the way down minimal size.
Where there's a common practice of ending macro defintions with '\n %{nil}', the parsed output of will become completely unfeasiable for use with scripts etc. that won't handle newline at at all kinds of unexpected places.

Ie. see [532a383a4bc19daee7ac96ce44947bfa12b0ad7c](https://github.com/rpm-software-management/rpm/pull/122/commits/532a383a4bc19daee7ac96ce44947bfa12b0ad7c), where without being able to shrink down the expanded macros, there would've been several newlines in place breaking use of parallel.

**Comment by ignatenkobrain** [comment]
honestly, I'm not sure how is that useful... But that's not up to me to decide that.

**Comment by n3npq** [comment]
@ignatenkobrain : you are correct that stink (or trim/rtrim or strip) isn't very useful for the common task of inserting strings into shell script lets: the shell is very forgiving of white space.

However, when using embedded %{lua:...} templates, or when using --queryformat, where macros are also expanded, extra whitespace or newlines can be a killer.

That is why %{shrink:...} was implemented.

(aside)
Note that if you are gonna mess with macros, then you might also consider stealing the macro-stack-to-tuple-list syntax.

An example perhaps shows what can be done best:
```
$ ./rpm -D'args foo' -D'args bar' -D'args baz' -E"%{@args:, }"
foo, bar, baz
```

**Comment by proyvind** [comment]
why remove attribution from commit message?

**Comment by ignatenkobrain** [comment]
superseded by #296 which fixes some style issue in code + adds test

**Comment by pmatilai** [comment]
Merged manually (commit 9a9eefc and 7fe183d) from #296, I added back the mention it's ported from rpm5.

**Pull request closed without merging**